daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/ImmArray.scala

```diff
@@ -72,7 +72,7 @@ final class ImmArray[+A] private (
   private def uncheckedGet(idx: Int): A = array(start + idx)
 
   /** O(1) */
-  def length: Int = len
+  val length: Int = len
```

Comment:
There is not point to use extra memory to cache O(1) operation, specially here where `len` is a `val` of the class.